### PR TITLE
#56411 do not suggest a fix for a import conflict in a macro

### PR DESCRIPTION
**Pull request by mockersf**
as a first fix, remove the suggestion when the conflict happens in a macro as it will be useless

### CODE REVIEW

File: src/test/ui/issues/issue_56411.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// compile-pass
+
+struct T {}
+
+fn main() {}
```
Comment (estebank) [inline src/test/ui/issues/issue_56411.rs]: Why is this file here?

Comment (estebank) [inline src/test/ui/issues/issue_56411.rs]: These files usually go in an `auxiliary` directory. I'm ok with this if @petrochenkov is too.

**Review by estebank — APPROVED** [review]
One nitpick, otherwise fine by me.

File: src/test/ui/issues/issue_56411.rs (hunk 1 above)
Comment (mockersf) [reply src/test/ui/issues/issue_56411.rs]: yes, but `// aux-build:` would allow to build it as an extern crate, not to use it as a module. And the `auxiliary` folder doesn't contain a `mod.rs` file, so you can't import a module from it...

Comment (petrochenkov) [reply src/test/ui/issues/issue_56411.rs]: Could you rename the file into something less subtle than "same file, but with underscore", e.g. `issue_56411_aux.rs`?

File: src/librustc_resolve/lib.rs [hunk 2]
```diff
@@ -5030,11 +5030,13 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
...
                 Ok(snippet),
                 NameBindingKind::Import { directive, ..},
-                _dummy @ false,
+                false,
+                false,
             ) = (
                 cm.span_to_snippet(binding.span),
                 binding.kind.clone(),
                 binding.span.is_dummy(),
+                binding.span.ctxt().outer().expn_info().is_some(),
```
Comment (petrochenkov) [inline src/librustc_resolve/lib.rs]: If you rely on `source.span.hi().0 - binding.span.lo().0` being a valid index in `binding.span`, then you should check at least the next conditions - `!source.span.is_dummy() && !binding.span.is_dummy() && !from_macro(source.span) && !from_macro(binding.span)`.

Comment (petrochenkov) [reply src/librustc_resolve/lib.rs]: Actually, directly checking that `source.span.hi().0 - binding.span.lo().0` doesn't underflow and results in a valid index into `cm` would be the most reliable alternative.
This way we certainly don't get an ICE, but also keep the diagnostics for "good" macros.

**Comment by Dylan-DPC** [comment]
ping from triage @mockersf waiting for your reply on the review comments

**Comment by TimNN** [comment]
Ping from triage @mockersf: Some changes have been requested on this PR. Do you think you'll be able to take care of those?

**Comment by petrochenkov** [comment]
Rebased and fixed in https://github.com/rust-lang/rust/pull/57908
